dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -141,6 +142,8 @@
      */
     private final List<Exporter<?>> exporters = new ArrayList<Exporter<?>>();
 
+    private static final ConcurrentHashMap<Integer, Integer> randomPortSet = new ConcurrentHashMap<>();
```

Comment:
Already has a static field `Map<String, Integer> RANDOM_PORT_MAP` to save random port ?  Why add a new cache ?